sentry-core/src/test/java/io/sentry/core/HubTest.kt

```diff
@@ -31,12 +38,42 @@ class HubTest {
         assertNotSame(scope, clone)
         assertEquals("extra", clone.extras["extra"])
         assertEquals("message", clone.breadcrumbs.first().message)
-        assertEquals("transaction", scope.transaction)
-        assertEquals("fingerprint", scope.fingerprint[0])
+        assertEquals("transaction", clone.transaction)
+        assertEquals("fingerprint", clone.fingerprint.first())
         assertEquals("tags", clone.tags["tags"])
         assertEquals("a@a.com", clone.user.email)
     }
 
+    @Test
+    @Ignore("it's a shallow copy and we need a deep-copy") // TODO: https://www.baeldung.com/java-deep-copy
```

Comment:
@bruno-garcia all the clones should be shallow or deep-copy? right now it returns a new instance of the object but it's fields are still the same.

Reply:
I want to say shallow is OK but for Scope data, if the value is not immutable, copying the reference would be a problem.

Reply:
@bruno-garcia ok so we already have a task on asana for it, default implementation of cloning on Java is shallow and it's also faulty if we make the fields `final`